File: can_make_arithmetic_progression_from_sequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        int minimum = INT_MAX;
        int maximum = INT_MIN;
        int n = arr.size();

        for (int num : arr) {
            minimum = std::min(minimum, num);
            maximum = std::max(maximum, num);
        }

        int diff = (maximum - minimum) / (n - 1);
        std::unordered_set<int> nums(arr.begin(), arr.end());

        if (diff == 0) {
            return nums.size() == 1;
        }

        for (int i = minimum; i <= maximum; i += diff) {
            if (nums.find(i) == nums.end()) {
                return false;
            }
        }

        return true;        
    }
};

File: can_make_arithmetic_progression_from_sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        minimum = min(arr)
        maximum = max(arr)
        n = len(arr)

        diff = (maximum - minimum) // (n - 1)
        nums = set(arr)

        if diff == 0:
            return len(nums) == 1
        
        for i in range(minimum, maximum + 1, diff):
            if i not in nums:
                return False
        
        return True
